go/merkletree2/tree.go

```diff
@@ -154,19 +174,69 @@ type PositionHashPair struct {
 	Hash     Hash     `codec:"h"`
 }
 
+type RootVersion uint8
+
+const (
+	RootVersionV1      RootVersion = 1
+	CurrentRootVersion RootVersion = RootVersionV1
+
+	RootVersionTesting RootVersion = 127
+)
+
+func (r RootVersion) GetEncodingType() EncodingType {
+	switch r {
+	case RootVersionV1:
+		return EncodingTypeBlindedSHA512_256v1
+	case RootVersionTesting:
+		return EncodingTypeForTesting
+	default:
+		panic(fmt.Sprintf("Unrecognized root version: %v", r))
+	}
+}
+
 type RootMetadata struct {
-	// TODO Add timestamp, version.....
 	_struct          struct{} `codec:",toarray"` //nolint
-	Seqno            Seqno    `codec:"n"`
-	BareRootHash     Hash     `codec:"r"`
-	SkipPointersHash Hash     `codec:"s"`
+	RootVersion      RootVersion
+	Seqno            Seqno
+	BareRootHash     Hash
+	SkipPointersHash Hash
+
+	//  AddOnsHash is the (currently empty) hash of a (not yet defined) data
+	//  structure which will contain a map[string]Hash (or even
+	//  map[string]interface{}) which can contain arbitrary values. This AddOn
+	//  struct is not used in verifying proofs, and new elements can be added to
+	//  this map without bumping the RootVersion. Clients are expected to ignore
+	//  fields in this map which they do not understand.
+	AddOnsHash Hash
```

Comment:
Is this encoded as `null` or an array of 32 `0x00`s ?

Reply:
Is there a test that everything still works if this is non-nil?

Reply:
Hash is of type []byte, not [32]byte. So I assume it is encoded as nil or as an empty slice, but definitely not as 32 `0x00`.
I can add a test, but I see no reason why this should fail as we are leveraging standard msgpack encoding here.